include/net/nrf_cloud.h

```diff
@@ -417,6 +418,64 @@ struct nrf_cloud_device_status {
 	struct nrf_cloud_svc_info *svc;
 };
 
+/** @brief GNSS data type contained in @ref nrf_cloud_gnss_data */
+enum nrf_cloud_gnss_type {
+	NRF_CLOUD_GNSS_TYPE_INVALID,
+	NRF_CLOUD_GNSS_TYPE_PVT,
+	NRF_CLOUD_GNSS_TYPE_NMEA,
+	NRF_CLOUD_GNSS_TYPE_MODEM_PVT,
```

Comment:
Why both `NRF_CLOUD_GNSS_TYPE_PVT` and `NRF_CLOUD_GNSS_TYPE_MODEM_PVT`?  Ditto for NMEA.

Reply:
maybe a bit overkill... but it allows for users to pass in the `nrf_modem_gnss_...` structs that might already be in use.
if not, the `nrf_cloud_gnss_...` can be used.  nrf cloud only uses some of the PVT fields.

Reply:
i'lll add some comments to make this more clear

Reply:
I figured you did that because the modem version is overkill.  Your code would be cleaner to just use the desired fields on the modem version and not have a separate structure, ID, and handling logic.  OTOH, the nrfcloud structure is more abstract and not coupled directly to modemlib.

Reply:
On the flip side, this does allow user code to be cleaner, since they will not have to create intermediary structs. I think there is some value in that

Reply:
i will leave this as is unless someone else objects.